pkg/git/gitlab/gitlab.go

```diff
@@ -197,6 +197,12 @@ func (g *Gitlab) getProject() error {
 	if err != nil {
 		return err
 	}
+	if project.Archived {
```

Comment:
This is dangerous. A get operation suggests to be read only. Having the unarchive operation within this function, could result in unexpected behavior.

I would prefer to have this function returning an error and the calling code taking care of calling unarchive if it makes sense.

Reply:
You're absolutely right. Thinking about it again, I don't think it's a good idea to un-archive a repo. I guess the current behavior of failing if the repo already exists in an archived state is ok.